### PR TITLE
update shape  from int to int64

### PR DESCRIPTION
**Pull request by seiriosPlus**
When using Large Distributed Lookup Table,  the variable's shape may exceed the upper limit of Integer, such as a  size of 100 billion word embedding.

### CODE REVIEW

File: paddle/fluid/framework/attribute.h [hunk 1]
```diff
@@ -235,7 +268,7 @@ class TypedAttrChecker {
     return *this;
   }
 
-  void operator()(AttributeMap& attr_map) const {
+  void operator()(AttributeMap& attr_map) const {  // NOLINT
```
Comment (jacquesqiao) [inline paddle/fluid/framework/attribute.h:271]: `const` AttributeMap& attr_map ？

File: paddle/fluid/framework/attribute.h [hunk 2]
```diff
@@ -271,7 +304,7 @@ class OpAttrChecker {
     return *(checker.target<TypedAttrChecker<T>>());
   }
 
-  void Check(AttributeMap& attr_map) const {
+  void Check(AttributeMap& attr_map) const {  // NOLINT
```
Comment (jacquesqiao) [inline paddle/fluid/framework/attribute.h:307]: `const` AttributeMap& attr_map?

File: paddle/fluid/framework/framework.proto [hunk 3]
```diff
@@ -25,16 +25,17 @@ message Version { optional int64 version = 1 [ default = 0 ]; }
...
-  LONG = 9;
-  BLOCKS = 10;
+  LONG = 1;
+  FLOAT = 2;
+  STRING = 3;
+  INTS = 4;
+  LONGS = 5;
+  FLOATS = 6;
+  STRINGS = 7;
+  BOOLEAN = 8;
```
Comment (jacquesqiao) [inline paddle/fluid/framework/framework.proto]: Do not change the order

**Comment by jacquesqiao** [comment]
Please add some background information about this chage.

File: paddle/fluid/framework/ir/graph.cc [hunk 4]
```diff
@@ -69,6 +69,12 @@ bool IsDistTrainOp(ir::Node *node, const std::vector<std::string> &send_vars,
           std::find(rpc_vars.begin(), rpc_vars.end(), var) != rpc_vars.end()) {
         return true;
       }
+
+      if (!(var.find(".block") == std::string::npos &&
```
Comment (Yancey0623) [inline paddle/fluid/framework/ir/graph.cc]: what's the difference between this condition statement and the above one? And please merge them and add some comments.

File: paddle/fluid/framework/framework.proto [hunk 5]
```diff
@@ -45,15 +46,16 @@ message OpDesc {
...
-    optional float f = 4;
-    optional string s = 5;
-    repeated int32 ints = 6;
-    repeated float floats = 7;
-    repeated string strings = 8;
-    optional bool b = 10;
-    repeated bool bools = 11;
-    optional int32 block_idx = 12;
-    optional int64 l = 13;
+    optional int64 l = 4;
```
Comment (Yancey0623) [inline paddle/fluid/framework/framework.proto]: Just append the new field at the tail and do not change the old field ID?

Comment (seiriosPlus) [reply paddle/fluid/framework/framework.proto]: done

File: paddle/fluid/framework/attribute.h (hunk 1 above)
Comment (seiriosPlus) [reply paddle/fluid/framework/attribute.h:271]: the attr_map is been changed in operator().

File: paddle/fluid/framework/attribute.h (hunk 2 above)
Comment (seiriosPlus) [reply paddle/fluid/framework/attribute.h:307]: ```attr_map``` will be modified in ```Check()```

**Pull request merged**